src/libbson/tests/test-json.c

```diff
@@ -1764,6 +1764,53 @@ test_bson_json_dbref (void)
                                           BCON_BOOL (true),
                                           "}");
 
+   const char *json_with_db = "{ \"key\": {"
+                              "\"$ref\": \"collection\","
+                              "\"$id\": 1,"
+                              "\"$db\": \"database\"}}";
+
+   bson_t *bson_with_db = BCON_NEW ("key",
+                                    "{",
+                                    "$ref",
+                                    BCON_UTF8 ("collection"),
+                                    "$id",
+                                    BCON_INT32 (1),
+                                    "$db",
+                                    BCON_UTF8 ("database"),
+                                    "}");
+
+   /* Note: the following examples are not valid DBRefs but are intended to test
+    * that the Extended JSON parser leaves such documents as-is. */
```

Comment:
These additional tests go along with the DBRef validation I removed from the extended JSON parser.